builtin/credential/radius/path_config.go

```diff
@@ -0,0 +1,214 @@
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "Comma-Separated list of policies to grant upon successful RADIUS aithentication of an unregisted user (default: emtpy)",
+			},
+			"dial_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "Number of seconds before connect timeouts (default: 10)",
+			},
+			"read_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "Number of seconds before response timeouts (default: 10)",
+			},
+			"nas_port": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Default:     10,
+				Description: "RADIUS NAS port field (default: 10)",
+			},
+		},
+
+		ExistenceCheck: b.configExistenceCheck,
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigRead,
+			logical.CreateOperation: b.pathConfigCreateUpdate,
+			logical.UpdateOperation: b.pathConfigCreateUpdate,
+		},
+
+		HelpSynopsis:    pathConfigHelpSyn,
+		HelpDescription: pathConfigHelpDesc,
+	}
+}
+
+// Establishes dichotomy of request operation between CreateOperation and UpdateOperation.
+// Returning 'true' forces an UpdateOperation, CreateOperation otherwise.
+func (b *backend) configExistenceCheck(req *logical.Request, data *framework.FieldData) (bool, error) {
+	entry, err := b.Config(req)
+	if err != nil {
+		return false, err
+	}
+	return entry != nil, nil
+}
+
+/*
+ * Construct ConfigEntry struct using stored configuration.
+ */
+func (b *backend) Config(req *logical.Request) (*ConfigEntry, error) {
+
+	storedConfig, err := req.Storage.Get("config")
+	if err != nil {
+		return nil, err
+	}
+
+	if storedConfig == nil {
+		// No user overrides, return default configuration
+		return nil, nil
+	}
+
+	var result ConfigEntry
+
+	if err := storedConfig.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (b *backend) pathConfigRead(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	cfg, err := b.Config(req)
+	if err != nil {
+		return nil, err
+	}
+	if cfg == nil {
+		return nil, nil
+	}
+
+	resp := &logical.Response{
+		Data: structs.New(cfg).Map(),
+	}
+	resp.AddWarning("Read access to this endpoint should be controlled via ACLs as it will return the configuration information as-is, including any secrets.")
+	return resp, nil
+}
+
+func (b *backend) pathConfigCreateUpdate(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	// Build a ConfigEntry struct out of the supplied FieldData
+	cfg, err := b.Config(req)
+	if err != nil {
+		return nil, err
+	}
+	if cfg == nil {
+		cfg = &ConfigEntry{}
+	}
+
+	host, ok := d.GetOk("host")
+	if ok {
+		cfg.Host = strings.ToLower(host.(string))
+	} else if req.Operation == logical.CreateOperation {
+		cfg.Host = strings.ToLower(d.Get("host").(string))
```

Comment:
For required parameters, we should give an error message here rather than getting the zero value of the input. Same would apply to other required parameters (secret, port, etc)